hail/hail/src/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -322,11 +322,12 @@ class ServiceBackend(
             jobGroup.job_group_id,
             Some(JobStates.Failed),
           )
-          assert(
-            failedEntries.nonEmpty,
-            s"Job group ${jobGroup.job_group_id} for batch ${batchConfig.batchId} failed, but no failed jobs found.",
-          )
-          val error = readPartitionError(root, failedEntries.head.head.job_id - startJobId)
+          val error = if (failedEntries.isEmpty)
+            new HailException(
+              s"Job group ${jobGroup.job_group_id} for batch ${batchConfig.batchId} failed, but no failed jobs found. The billing project may be low on funds."
+            )
```

Comment:
Thanks for looking into this. In this case, I think we should test to see if any jobs have been cancelled and throw a cancellation exception if there are any. If there are none, we should throw an error saying go look at the ui. I don't think we can infer that there are insufficient funds in the billing project.

Reply:
The issue is that the job group wasn't cancelled, it failed. But none of the contained jobs failed, most were cancelled, perhaps because the billing project was low on funds.

A user reported hitting this assertion, so clearly it's making an invalid assumption.

@ehigham My thought about the insufficient funds note was just to hint that they may want to double check that. I've now seen that be the cause of mysterious failures twice, and there are no hints that that could be the cause. But certainly that should be a change to batch itself to surface that information.